esmvaltool/utils/cmorizers/obs/nsidc_common.py

```diff
@@ -0,0 +1,97 @@
+"""Common tools to CMORize NSIDC-0116 northern and sothern data."""
+
+import logging
+import os
+import glob
+import numpy as np
+from numba import vectorize
+import iris
+from iris.coords import AuxCoord
+from iris.cube import Cube
+
+
+from .utilities import fix_var_metadata, save_variable, set_global_atts
+
+logger = logging.getLogger(__name__)
+
+
+def cmorize(cfg, region, in_dir, out_dir):
+    """Cmorize NSIDC-0116 dataset."""
+    glob_attrs = cfg['attributes']
+
+    logger.info("Starting cmorization for Tier%s OBS files: %s",
+                glob_attrs['tier'], glob_attrs['dataset_id'])
+    logger.info("Input data from: %s", in_dir)
+    logger.info("Output will be written to: %s", out_dir)
+
+    file_expr = os.path.join(in_dir, 'icemotion_daily_{}_*.nc'.format(region))
+    for filepath in glob.glob(file_expr):
+        logger.info('Cmorizing file %s', filepath)
+        cubes = iris.load(filepath)
+        logger.debug(cubes)
+        lat_coord = _create_coord(cubes, 'lat', 'latitude')
+        lon_coord = _create_coord(cubes, 'lon', 'longitude')
+        lon_coord.points = _correct_lons(lon_coord.points)
+
+        for var, vals in cfg['variables'].items():
+            logger.info('Cmorizing var %s', var)
+            cube = cubes.extract_strict(iris.Constraint(vals['raw']))
+            cube.add_aux_coord(lat_coord, (1, 2))
+            cube.add_aux_coord(lon_coord, (1, 2))
+            logger.debug(cube)
+            var_info = cfg['cmor_table'].get_variable(vals['mip'], var)
+            glob_attrs['mip'] = vals['mip']
+            fix_var_metadata(cube, var_info)
+            set_global_atts(cube, glob_attrs)
+            zlib = vals.get('compress', False)
+            if zlib:
+                # Realize data to speed-up writing
+                # pylint: disable=pointless-statement
+                cube.data
+            save_variable(cube, var, out_dir, glob_attrs, zlib=zlib)
+            cubes.remove(cube)
+
+    _create_areacello(cfg, cube, glob_attrs, out_dir)
+
+
+@vectorize
```

Comment:
Could you use numpy instead? I would like to try and get rid of our numba dependency at some point.
I think you could do something like `lon[lon < 0] += 360`

Reply:
Numba is way faster. Why do you want to remove it?

Reply:
Interesting, how much time does this take with numpy compared to numba?

As we discussed at the workshop, I think it's fine to keep numba if needed for diagnostics, but at the moment I see no real need for it and it would be good if we could reduce the number of dependencies.

Reply:
> Interesting, how much time does this take with numpy compared to numba?

In a similar case it took me more than 10 seconds using numpy vectorize and less than half a second using numba. The code I used for those tests:

Numpy version
```python
    def calculate_weight(e3t, depth, mask):
        """
        Calculates the weight for each cell
        """
        if not mask:
            return 0
        top = depth
        bottom = top + e3t
        if bottom < min_depth or top > max_depth:
            return 0
        else:
            if top < min_depth:
                top = min_depth
            if bottom > max_depth:
                bottom = max_depth

            return (bottom - top) * 1020 * 4000

    calc = np.vectorize(calculate_weight, otypes='f')

```

Numba implementation
```python
@vectorize(['float32(int32, int32, float32, float32, float32)'], target='cpu')
def _calculate_weight(min_depth, max_depth, e3t, depth, mask):
    if not mask:
            return 0
    top = depth
    bottom = top + e3t
    if bottom < min_depth or top > max_depth:
            return 0
    else:
            if top < min_depth:
                    top = min_depth
            if bottom > max_depth:
                    bottom = max_depth

         return (bottom - top) * 1020 * 4000
```

Reply:
That is not a very fair comparison, with numpy you should use array slicing instead of looping over individual elements.

The [numpy.vectorize](https://docs.scipy.org/doc/numpy/reference/generated/numpy.vectorize.html) documentation warns against using this function if performance is required:
>  The vectorize function is provided primarily for convenience, not for performance. The implementation is essentially a for loop.